### DISCUSSION TITLE
How to check if field turn from inValid to valid onChange

### DISCUSSION THREAD

**Discussion by dandin84** [react-hook-form/react-hook-form#3967] [Q&A]
Hi,

After first submit error I succeeded to catch if there is an error:

![image](https://user-images.githubusercontent.com/6383184/105177378-9861d300-5af4-11eb-8ea0-435d5bd3d106.png)

So on error I can display 'X'

![image](https://user-images.githubusercontent.com/6383184/105177525-c9420800-5af4-11eb-802d-d328e0bf7c25.png)

Now I would like onChange to change the 'X' to 'V' when the field turn from inValid to Valid.

So it will look like:

![image](https://user-images.githubusercontent.com/6383184/105177975-7157d100-5af5-11eb-90f4-3245b97ded5d.png)

If success is true:

![image](https://user-images.githubusercontent.com/6383184/105178445-0955ba80-5af6-11eb-8ee3-3b457aa429ab.png)

But I could not find the right check which will give me success true only after inValid change to Valid and not on first load.

What would you suggest should be the right condition?

I am sure I just missing something :(

**Comment by bluebill1049** [comment]
pretty sure you can subscribe to the errors object plus dirty state.

**Reply by dandin84** [reply]
Yeah, thats what I was thinking too, just I could not find how to subscribe to the dirty state in the right way (it seems to be always false) but I will take another look at the docs than, thanks!
